options_test.go

```diff
@@ -395,3 +395,17 @@ func TestCustomSerialization(t *testing.T) {
 	require.Equal(t, 202, w.Code)
 	require.Equal(t, "custom serialization", w.Body.String())
 }
+
+func TestGroupParams(t *testing.T) {
+	s := NewServer()
+	group := Group(s, "/api").Header("X-Test-Header", "test-value")
+
+	Get(group, "/test", controller)
+	Get(group, "/test2", controller)
+
+	require.Equal(t, group.params, []GroupParams{{Name: "X-Test-Header", Description: "test-value", OpenAPIParam: OpenAPIParam{Required: false, Example: "", Type: "header"}}})
+
+	document := s.OutputOpenAPISpec()
+	require.Equal(t, document.Paths.Find("/api/test").Get.Parameters[0].Value.Name, "X-Test-Header")
+	require.Equal(t, document.Paths.Find("/api/test2").Get.Parameters[0].Value.Name, "X-Test-Header")
+}
```

Comment:
Thanks for thinking about tests!

I would add more tests

- query parameters
- verify it doesn't impact sibling groups / parent groups / children groups / main server instance
- what happens if I add `Hide()` method on the group? Does it panics?